Some refactoring on post-build scripts & templates

JohnTortugo opened the pull request:
Relates to #3188 

- Most of these changes are actually accounting for variable renaming. I created some new variables in `common-variables.yml` to encapsulate access to Maestro PROD/INT.
- Deleted `dotnetsymbol-init.ps1` and `sourcelink-cli-init.ps1` and inlined their content into `symbols-validation.ps1` and `sourcelink-validation.ps1`.
- Test build here: https://dnceng.visualstudio.com/internal/_build/results?buildId=278652&view=results activated all four channels and published to test feeds. Executed in INT.

riarenas commented on eng/common/templates/post-build/common-variables.yml:
```diff
@@ -1,21 +1,39 @@
+
   # .NET Core 3 Dev
-  PublicDevRelease_30_Channel_Id: 3
+  - name: PublicDevRelease_30_Channel_Id
```
What's the reason for changing the key: value syntax?

JohnTortugo replied:
I included the `Publish-Build-Assets` group (line 2) in the variable template and after that AzDO started complaining about the syntax of the simplified variable declaration. My take was that I couldn't mix group linking and simplified variable declaration in the same set of variable declaration.

riarenas replied:
That's pretty much what I imagined

---
In reply to: [307858289](https://github.com/dotnet/arcade/pull/3385#discussion_r307858289) [](ancestors = 307858289)

riarenas commented on eng/common/post-build/trigger-subscriptions.ps1:
```diff
@@ -1,33 +1,20 @@
-  [string] $BarToken,
-  [string] $ApiVersion = "2019-01-16"
+  [Parameter(Mandatory=$true)][string] $MaestroApiAccessToken,
+  [Parameter(Mandatory=$false)][string] $MaestroApiEndPoint = "https://maestro-prod.westus2.cloudapp.azure.com",     # Maestro API URL
```
I think the comments are unnecessary for these two, or lets add the comments to all of them

riarenas commented on eng/common/post-build/sourcelink-validation.ps1:
```diff
@@ -3,13 +3,11 @@ param(
   [Parameter(Mandatory=$true)][string] $GHCommit,               # GitHub commit SHA used to build the packages
-  [Parameter(Mandatory=$true)][string] $SourcelinkCliVersion    # Version of SourceLink CLI to use
+  [Parameter(Mandatory=$true)][string] $SourcelinkCliVersion,   # Version of SourceLink CLI to use
+  [switch] $JustSetup                                           # If set the script will just install sourcelink CLI
```
Is this being used?

JohnTortugo replied:
`JustSetup` is used on line 221.

riarenas replied:
Yes, but is anyone calling it with the switch set? What is the scenario to only install the cli and not actually do anything with it?

---
In reply to: [307866766](https://github.com/dotnet/arcade/pull/3385#discussion_r307866766) [](ancestors = 307866766)

JohnTortugo replied:
Got your point. I guess I just followed suit what the `eng\common\darc-init.ps1` did and that doesn't make a lot of sense for these scripts. I'll remove that. Thanks Ricardo!

JohnTortugo replied:
Removed that

riarenas commented on eng/common/post-build/post-build-utils.ps1:
```diff
@@ -0,0 +1,55 @@
+# Most of the functions in this file require the variables `MaestroApiEndPoint`, 
+# `MaestroApiVersion` and `MaestroApiAccessToken` to be globally available.
```
>MaestroApiVersion` and `MaestroApiAccessToken` to be globally available [](start = 3, length = 71)

Should we error out if this is not the case? instead of waiting for the inner error or whatever?

JohnTortugo replied:
I implemented some validation. Please see `post-build-utils.ps1` -> Validate-MaestroVars

riarenas commented:
LGTM with some comments.

Let's make sure to run another test build like the one you did before merging.

JohnTortugo commented:
@riarenas can you PTAL again? I addressed your comments and also created another PS script: `setup-maestro-vars.ps1` which is called from `setup-maestro-vars.yml`.

Test build is here: https://dnceng.visualstudio.com/internal/_build/results?buildId=281593&view=results

Looks good to merge?

riarenas commented:
LGTM
